### DISCUSSION TITLE
2FA Must be OPTIONAL

### DISCUSSION THREAD

**Discussion by AwokeKnowing** [community/community#70864] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Product Feedback

### Body

github grew to be the Open Source community hub.  In open source, we are all adults and we make OUR OWN CHOICES. particularly and passwords and such.

It is 100% against the open source ethos to force 2fa which requires external dependencies to just access your code.

This is FAR BEYOND THE RED LINE.  

Microsoft are now just become bullies and made Github another Microsoft app. 

Microsoft NEEDS TO LEARN why they always seem to be winning and then miserably fail.  It is always the same pattern of some stupid project manager unfit for their job FORCING some policy which is against users making their own choices. That makes microsoft unfit to host the open source community's work. Now they don't even let you decide how to authenticate. they force you to tie your account to SMS or some app. 

Forcing 2fa against dev's will is EXACTLY the kind of behavior that made MS infamous. why don't you get it? MAKE 2FA OPTIONAL

**Comment by yutent** [comment]
2FA is a pile of shit

**Comment by VIPLightning** [comment]
This is Microsoft's doing and 2FA adds potentially greater security risk. That's pretty much what's happened to 100 million epic games users that were hacked by the Russians. But I digress. I'd switch and migrate all my work to [Codeberg](https://codeberg.org/) or other git websites before December 15 at 0:00 (UTC) arrives and shit hits the fan. Because when it happens, Microsoft pretty much made this website a high valuable target to hit and Microsoft's gonna do what's only in their best interest and not for consumers or developers.
